File: onlinepayments-sdk/src/main/java/com/onlinepayments/sdk/client/android/model/paymentproduct/AccountOnFileDisplay.java
/*
 * Copyright 2020 Global Collect Services B.V
 */

package com.onlinepayments.sdk.client.android.model.paymentproduct;

import java.io.Serializable;


/**
 * POJO that represents an AccountOnFile object.
 */
public class AccountOnFileDisplay implements Serializable {

	private static final long serialVersionUID = -7793293988073972532L;

	private String attributeKey;
	private String mask;

	/**
	 * @deprecated In a future release, this constructor will become internal to the SDK.
	 */
	@Deprecated
	public AccountOnFileDisplay(String attributeKey, String mask) {
		this.attributeKey = attributeKey;
		this.mask = mask;
	}

	public String getKey() {
		return attributeKey;
	}

	public String getMask() {
		return mask;
	}

}

File: onlinepayments-sdk/src/main/kotlin/com/onlinepayments/sdk/client/android/model/paymentproduct/AccountOnFileDisplay.kt
/*
 * Do not remove or alter the notices in this preamble.
 *
 * Copyright © 2025 Worldline and/or its affiliates.
 *
 * All rights reserved. License grant and user rights and obligations according to the applicable license agreement.
 *
 * Please contact Worldline for questions regarding license and user rights.
 */
package com.onlinepayments.sdk.client.android.model.paymentproduct

import java.io.Serializable

/**
 * Data class that represents an AccountOnFile object.
 */
class AccountOnFileDisplay(
    val attributeKey: String?,
    val mask: String?
) : Serializable {

    fun getKey(): String? {
        return attributeKey
    }

    companion object {
        @Suppress("Unused")
        private val serialVersionUID = -7793293988073972532L
    }
}
